readme.txt
===BP Group Documents  ===
Contributors: lenasterg
Tags: wpms, buddypress, group, document, plugin, file, media, storage, upload, widget
Requires at least: WP 3.5, BuddyPress 1.7
Tested up to: 3.6, BuddyPress 1.8
Stable tag: 1.2 (Requires at least: WP 3.5, BuddyPress 1.7)
License: GNU General Public License 3.0 or newer (GPL) http://www.gnu.org/licenses/gpl.html
Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=Q4VCLDW4BFW6L 

BP Group Documents creates a page within each BuddyPress group to upload and any type of file or document. 

== Description ==
Original plugin is <a href="http://wordpress.org/extend/plugins/buddypress-group-documents/">no longer supported</a> so revised.

BP Group Documents creates a page within each BuddyPress group to upload and any type of file or document. This allows members of BuddyPress groups to upload and store files and documents that are relevant to the group.

Documents can be edited and deleted either by the document owner or by the group administrator.
Categories can be used to organize documents. 
Activity is logged in the main activity stream, and is also tied to the user and group activity streams.
The site administrator can set filters on file extensions, set display options, and upload files via FTP.
Group members and moderators can receive email notifications at their option. 
The group administrator can decide if all members or only admins/moderators can upload documents (Since v0.5)
User verification for Downloads: when a document is downloaded, a redirect page checks is the user is member of the group (in case of a private  or hidden groups) and only then the user can download the file.(Since v0.5)
3 Widgets: "User's groups documents" (since v0.5) ,"Recent Uploads" , "Popular Downloads"  can be used to show activity at a glance.
Contributions by Lena Stergatu for WP 3.3, with additional bug fixes and improvements by Keeble Smith (http://keeblesmith.com) and Anton Andreasson work for BP 1.7. Original plugin author Peter Anselmo.

PLEASE: If you have any issues or it doesn't work for you, please report in support forum.  It doesn't help anyone to mark "broken" without asking around.  Thanks!  

NOTE: For Buddypress v1.5, use the plugin's version 0.4.3.3. available in http://lenasterg.wordpress.com/2012/04/17/buddypress-group-documents-for-bp-1-5-and-wp-3-3/

== Installation ==

Make sure Wordpress and BuddyPress are installed and active.

Copy the plugin folder buddypress-group-documents/ into /wp-content/plugins/

Browse to the plugin administration screen and activate the plugin.

There will now be a "Group Documents" menu item under the "Settings" menu.  Here you will find a list of all file extensions allowed for uploaded files along with other settings.

== Upgrade Notice == 
If you upgrade from older version you must also add a  your .htaccess in order 
to ensure that requests to the old URLs are redirected to the new, hardened URL
That line is: 
RewriteRule ^wp\-content/blogs\.dir/1/files/group\-documents/(.*) /?get_group_doc=$1 [R,L]


== Frequently Asked Questions ==
 1. If you run a windows server and you get errors about mb_convert_case  function which is a default php function (see http://php.net/manual/en/function.mb-convert-case.php), you must uncomment the line with php_mbstring.dll in your php.ini.
 2. If you are a plugin developer and want to use the upload file form you can link to /group_slug/bpgroupdocuments_slug/add to access the upload document form

== Screenshots ==
1. Admin settings page
2. Documents settings page on group creation
3. Upload document form
4. Document list tab
5. Ties into site activity stream (for public groups only)
6. Group admin document's settings tab, allow category's edit, addition 
7. User options for email notifications
8. Widget Recent Documents from your groups  and Widget Popular Group Documents
9. Message when non member of a private or hidden group tries to access a group document


==Changelog==
Version 1.2 (September 3,2013)
- Added Swedish translation. Thanks goes to nat0n (http://wordpress.org/support/profile/nat0n)

Version 1.1 (September 3,2013)
- Fix some broken links by changing the plugin directory name with BP_GROUP_DOCUMENTS_DIR constant

Version 1.0 (August 28,2013)
- Update readme.txt
- Add screenshots

See history.txt for older version changelog
Apologies for the frequent updates, this plugin is under active development!

== Notes ==

Roadmap.txt - contains ideas proposed and the (approximate) order of implementation

History.txt - contains all the changes since version .1

License.txt - contains the licensing details for this component.

== Feedback ==

Please leave a comment  with any bugs, improvements, or comments at http://lenasterg.wordpress.com


== Donate ==
You can  say "Thank You" via my Amazon wish list
http://www.amazon.co.uk/registry/wishlist/7HYK62UCVCDI
or donate at paypal https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=Q4VCLDW4BFW6L 

history.txt
Version 1.1 (September 3,2013)
- Fix some broken links by changing the plugin directory name with BP_GROUP_DOCUMENTS_DIR constant

Version 1.0 (August 28,2013)
- Update readme.txt
- Add screenshots

Version 0.9 (August 27,2013)
- Update all language files 
- Add localazition support to javascript messages
- Fix the message strings

Version 0.8 (August 5,2013)
- New feature: Now it is possible to link from other plugins  to  /groups/nameofgroup/documentsslug/add. The upload form is displayed open. 
- Now BP_Group_Documents ->current_user_can($action, $group_id) takes also group_id variable. If no group_id is provided it uses current group
- Change visibility of documents tab to private


Version 0.7 (June 21, 2013)
- Fix a bug which prevented the notifications settings to be saved

Version 0.6 (June 4, 2013)
- Add settings link on plugin page
- Fix the admin menu link for single wp installation

Version 0.5.4 (May 21, 2013)
- Add documents categories info into the documents loop
- Fix the edit,delete category from group admin bug
- Uses user defined BP Group documents title in various places (widgets, activity filter etc)
- Add option for show/hide file downloads count
- Add background icon for #nav-documents-groups-li into style.css

Version 0.5.3 Beta(May 10, 2013)
 - Delete a debug typo


Version 0.5.2 Beta(May 10, 2013)
- Change Content-Disposition to inline in order to allow document to open into browser or be downloaded 

Version 0.5.1 Beta(May 3, 2013)
- Fix a bug of widgets.php

Version 0.5 Beta(May 1, 2013)
- By Stergatu Lena (http://lenasterg.wordpress.com)
- Compatible with Buddypress 1.7
- Added creation step on group creation
- Added user group documents widget
- Various deprecated code removed and rewrites haven 
- Minor fixes at style.css
- New strings in .pot file



Version 0.4.3.3 (May 13, 2012)
- By Keeble Smith Ltd (http://keeblesmith.com)
- Fixed layout of sidebar (early closing of div) in some configurations

Version 0.4.3.2 (May 8, 2012)
-  By Lena Stergatu stergatu@cti.gr
- Minor fixes at style.css
- New strings in .pot file

Version 0.4.3.1 (May 5, 2012)
- This update by Keeble Smith Ltd (http://keeblesmith.com) - thanks to Lena Stergatu for adopting (stergatu@cti.gr)
- *** Please Note:
- *** This update includes a complete rewrite of style.css to allow themes to override correctly, simplify, create clearer element identification and remove obsolete css
- *** You may need to update any css overrides you are implementing
- *** This update also includes major changes to the form layout and other HTML elements which may cause require additional css changes, but should allow future updates to be more stable
- Added BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR default setting (false)
- Added conditional use of wp_editor, based on BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR setting (to provide theme authors with some control - ideally should be a setting.
- Added logical check of BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR so description is stored correctly either way
- Added logical check of BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR in do_post_logic for description property population in add and edit contexts
- Updated get_file_size to test file_exists, avoiding warning messages being written to the error_log
- Fixed category filter disappears if no documents exist in the selected category, making it impossible to then change the filter back and see any documents.
- Multiple changes to the form layout and styling - simpler, cleaner HTML and CSS which now validates and is compatible with more themes:
-- Moved Document List heading so it will appear even if there are no documents to show
-- Updated document description display to use stripslashes if using wp_editor and nl2br if not
-- Updated bp-group-documents-delete to a class instead of ID (avoiding duplicate IDs in HTML)
-- Removed obsolete spacer div (using css for styling instead)
-- Corrected immediate closing of bp-group-documents-form tag (invalid HTML)
-- Fixed layout of headings and categories list being bunched up
-- Added a margin at the bottom of bp-group-documents div
-- When no documents exist in a group the 'upload a new document' button no longer overlaps the 'There are no documents...' message in some themes
-- Only output the existing categories list if there are any existing categories
-- Moved submit button to below the categories list (for easier styling customisation)
- Multple invalid HTML fixes and cleanup:
-- Removed unnecessary br tags (use CSS for styling)
-- Added <p> tag around the Valid File Formats text.
-- Added rows and cols attributes to bp-group-documents-description textarea (required for valid HTML)
-- Removed empty label tag
-- Replaced unclosed BR tags with closed BR tags
-- Renamed group-documents-edit-link target from #e to #edit-document-form to remove ambiguity
-- Updated group-documents-icon to bp-group-documents-icon
-- Renamed document-info div to bp-group-documents-document-info for unambiguous & consistent identity
- Added verion caching control for general.js so sites updating plugin will get latest version of the JavaScript
- general.js - updated to adopt changes made from ID to class, and naming consistency

Aplogies for the frequent updates, this plugin is under active development!
Version 0.4.3 (May 2, 2012)
- Add activity dropdown filtering
- Add option for administrators to rename the group documents navigation tab
-Add some strings in .pot
-Add en_US.po file in order to allow developers to rename the English strings 


Version 0.4.2 (April 24, 2012)
-  By Lena Stergatu stergatu@cti.gr
- Add wp_editor support
- Fix the style and js script position 
- Added some localized strings

Version 0.4.1 (April 17, 2012)
----------------------------------------------------------
- Bug fix


Version 0.4 (April 2, 2012)
----------------------------------------------------------

1.2/loader.php
<?php

/*
  Plugin Name: BP Group Documents
  PLugin URI: http://lenasterg.wordpress.com/
  Description: BP Group Documents creates a page within each BuddyPress group to upload and any type of file or document. 
  Version: 1.2
  Revision Date: September 3, 2013
  Requires at least: WP 3.5.1, BuddyPress 1.7
  Tested up to: WP 3.6, BuddyPress 1.8
  License:  GNU General Public License 3.0 or newer (GPL) http://www.gnu.org/licenses/gpl.html
  Author: Lena Stergatu (http://lenasterg.wordpress.com) since v. 0.4.3.3. with additional bug fixing and improvements by <a href="http://keeblesmith.com">Keeble Smith</a>. Original plugin author Peter Anselmo, Studio66.
  Author URI:
  Network Only: true
 * @todo minor, make a deregister function, 26/4/2013 stergatu
 */


/* Only load code that needs BuddyPress to run once BP is loaded and initialized. */

//some constants that can be checked when extending this plugin
define('BP_GROUP_DOCUMENTS_IS_INSTALLED', 1);
define('BP_GROUP_DOCUMENTS_VERSION', '0.5');
define('BP_GROUP_DOCUMENTS_DB_VERSION', '5');
define('BP_GROUP_DOCUMENTS_VALID_FILE_FORMATS', 'odt,rtf,txt,doc,docx,xls,xlsx,ppt,pps,pptx,pdf,jpg,jpeg,gif,png,zip,tar,gz');
define('BP_GROUP_DOCUMENTS_ITEMS_PER_PAGE', 20);
define('BP_GROUP_DOCUMENTS_UPLOAD_PERMISSION', 'mods_decide');
define('BP_GROUP_DOCUMENTS_DISPLAY_ICONS', 1);
define('BP_GROUP_DOCUMENTS_USE_CATEGORIES', 1);
define('BP_GROUP_DOCUMENTS_ENABLE_ALL_GROUPS', true);
define('BP_GROUP_DOCUMENTS_PROGRESS_BAR', 1);
define('BP_GROUP_DOCUMENTS_FORUM_ATTACHMENTS', 0);


//allow override of URL slug
if (!defined('BP_GROUP_DOCUMENTS_SLUG')) {
    define('BP_GROUP_DOCUMENTS_SLUG', 'documents');
}

/**
 * Nifty function to get the name of the directory bp_group_documents plugin is installed in.
 * @author  Stergatu Eleni 
 * @version 1, 6/3/2013
 * @since 0.5
 */
function bp_group_documents_dir() {
    if (stristr(__FILE__, '/'))
        $bp_gr_dir = explode('/plugins/', dirname(__FILE__));
    else
        $bp_gr_dir = explode('\\plugins\\', dirname(__FILE__));
    return str_replace('\\', '/', end($bp_gr_dir)); //takes care of MS slashes
}

$bp_gr_dir = bp_group_documents_dir();

define('BP_GROUP_DOCUMENTS_DIR', $bp_gr_dir); //the name of the directory that bp_group_documents  files are located.


/**
 * @author Stergatu Eleni 
 * @global type $wpdb
 * @return type
 * @since 0.5
 * @version 1 30/4/2013
 */
function bp_group_documents_init() {
    global $wpdb;
    if (is_multisite() && BP_ROOT_BLOG != $wpdb->blogid)
        return;
    if (!bp_is_active('groups'))
        return;
    // Because our loader file uses BP_Component, it requires BP 1.5 or greater.
    if (version_compare(BP_VERSION, '1.6.5', '>')) {
        bp_group_documents_set_constants();
        require( dirname(__FILE__) . '/buddypress-group-documents.php' );
        bp_group_documents_include_files();
    }
}

add_action('bp_include', 'bp_group_documents_init');

/**
 * bp_group_documents_is_installed()
 * Checks to see if the DB tables exist or if we are running an old version
 * of the component. If the value has increased, it will run the installation function.
  @since 0.5
 * @version 1, 5/3/2013
 */
function bp_group_documents_is_installed() {
    bp_group_documents_set_constants();
    if (get_site_option('bp-group-documents-db-version') < BP_GROUP_DOCUMENTS_DB_VERSION) {
        bp_group_documents_install_upgrade();
    }
}

register_activation_hook(__FILE__, 'bp_group_documents_is_installed');

/**
 * bp_group_documents_install_upgrade()
 *
 * Installs and/or upgrades the database tables
 * This will only run if the database version constant is         
 * greater than the stored database version value or no database version found
 * @author Stergatu Eleni <stergatu@cti.gr>
 * @version 1.0, 7/3/2013 now uses add_site_option instead of add_option 
 * @since 0.5

 */
function bp_group_documents_install_upgrade() {
    global $wpdb, $bp;
      
    $charset_collate = '';
    if (!empty($wpdb->charset)) {
        $charset_collate = "DEFAULT CHARACTER SET $wpdb->charset";
    }

    //If there is a previous version installed then move the variables to the sitemeta table
    if ((get_site_option('bp-group-documents-db-version')) && (get_site_option('bp-group-documents-db-version') < 5)) {
        $sql[] = bp_group_documents_tableCreate($charset_collate);
    }
    if (!get_site_option('bp-group-documents-db-version')) {
        $sql[] = bp_group_documents_tableCreate($charset_collate);
        add_option('bp_group_documents_nav_page_name', __('Documents', 'bp-group-documents'));
        add_option('bp_group_documents_valid_file_formats', BP_GROUP_DOCUMENTS_VALID_FILE_FORMATS);
        add_option('bp_group_documents_items_per_page', BP_GROUP_DOCUMENTS_ITEMS_PER_PAGE);
        add_option('bp_group_documents_upload_permission', BP_GROUP_DOCUMENTS_UPLOAD_PERMISSION);
        add_option('bp_group_documents_display_icons', BP_GROUP_DOCUMENTS_DISPLAY_ICONS);
        add_option('bp_group_documents_use_categories', BP_GROUP_DOCUMENTS_USE_CATEGORIES);
        add_option('bp_group_documents_enable_all_groups', BP_GROUP_DOCUMENTS_ENABLE_ALL_GROUPS);
        add_option('bp_group_documents_progress_bar', BP_GROUP_DOCUMENTS_PROGRESS_BAR);
        add_option('bp_group_documents_forum_attachments', BP_GROUP_DOCUMENTS_FORUM_ATTACHMENTS);
        add_option('bp-group-documents-db-version', BP_GROUP_DOCUMENTS_DB_VERSION);
    }
    update_site_option('bp-group-documents-db-version', BP_GROUP_DOCUMENTS_DB_VERSION);

    require_once( ABSPATH . "wp-admin/includes/upgrade.php" );


    dbDelta($sql);
}

/**
 * SQL create command for BP_GROUP_DOCUMENTS_TABLE 
 * @since version 0.5
 * @author stergatu
 * @version 1.0, 25/4/2013
 * @param type $charset_collate
 * @return string
 */
function bp_group_documents_tableCreate($charset_collate) {
    $toSql = $sql[] = "CREATE TABLE " . BP_GROUP_DOCUMENTS_TABLE . " (
		  		id bigint(20) NOT NULL AUTO_INCREMENT PRIMARY KEY,
		  		user_id bigint(20) NOT NULL,
		  		group_id bigint(20) NOT NULL,
		  		created_ts int NOT NULL,
				modified_ts int NOT NULL,
				file VARCHAR(255) NOT NULL,
				name VARCHAR(255) NULL,
				description TEXT NULL,
				featured BOOL DEFAULT FALSE,
				download_count bigint(20) NOT NULL DEFAULT 0,
                                KEY user_id (user_id),
                                KEY group_id (group_id),
                                KEY created_ts (created_ts),
				KEY modified_ts (modified_ts),
				KEY download_count (download_count)
		 	   ) {$charset_collate};";
    return $toSql;
}

/**
 * @author Stergatu Eleni  
 * @since 0.5
 * @version 2, 24/4/2013
 */
function bp_group_documents_set_constants() {
    global $wpdb, $bp;
    if (!defined('BP_GROUP_DOCUMENTS_TABLE'))
        define('BP_GROUP_DOCUMENTS_TABLE', $wpdb->base_prefix . 'bp_group_documents');


    //This is where to look for admin bulk uploads
    if (!defined('BP_GROUP_DOCUMENTS_ADMIN_UPLOAD_PATH'))
        define('BP_GROUP_DOCUMENTS_ADMIN_UPLOAD_PATH', WP_PLUGIN_DIR .'/'. BP_GROUP_DOCUMENTS_DIR. '/uploads/');

    //Widgets can be set to only show documents in certain (site-admin specified) groups
    if (!defined('BP_GROUP_DOCUMENTS_WIDGET_GROUP_FILTER'))
        define('BP_GROUP_DOCUMENTS_WIDGET_GROUP_FILTER', true);

    //if enabled, documents can be flagged as "featured"
    //widget will have an option to only show featured docs
    if (!defined('BP_GROUP_DOCUMENTS_FEATURED'))
        define('BP_GROUP_DOCUMENTS_FEATURED', true);

    //longer text descriptions to go with the documents can be toggled on or off.
    //this will toggle both the textarea input, and the display;
    if (!defined('BP_GROUP_DOCUMENTS_SHOW_DESCRIPTIONS'))
        define('BP_GROUP_DOCUMENTS_SHOW_DESCRIPTIONS', true);

    //if enabled, and wp_editor exists, it will be used for the document comment editor
    if (!defined('BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR'))
        define('BP_GROUP_DOCUMENTS_ALLOW_WP_EDITOR', false);

    switch (substr(BP_VERSION, 0, 3)) {
        case '1.1':
            define('BP_GROUP_DOCUMENTS_THEME_VERSION', '1.1');
            break;
        case '1.2':
        default:
            if ('BuddyPress Classic' == wp_get_theme()) {
                define('BP_GROUP_DOCUMENTS_THEME_VERSION', '1.1');
            } else {
                define('BP_GROUP_DOCUMENTS_THEME_VERSION', '1.2');
            }
            break;
    }
    do_action('bp_group_documents_globals_loaded');
}

?>